pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -965,37 +969,43 @@ self: super: {
   hledger = overrideCabal super.hledger (drv: {
     postInstall = ''
       for i in $(seq 1 9); do
-        for j in $data/share/${self.ghc.name}/${pkgs.stdenv.system}-${self.ghc.name}/*/*.$i $data/share/${self.ghc.name}/${pkgs.stdenv.system}-${self.ghc.name}/*/.otherdocs/*.$i; do
+        for j in $data/share/${self.ghc.name}/${ghc_system}-${self.ghc.name}/*/*.$i $data/share/${self.ghc.name}/${ghc_system}-${self.ghc.name}/*/.otherdocs/*.$i; do
           mkdir -p $out/share/man/man$i
           cp $j $out/share/man/man$i/
         done
       done
-      mkdir $out/share/info
-      cp $data/share/${self.ghc.name}/${pkgs.stdenv.system}-${self.ghc.name}/*/*.info $out/share/info/
+      for i in $data/share/${self.ghc.name}/${ghc_system}-${self.ghc.name}/*/*.info; do
+        mkdir -p $out/share/info
+        cp $i $out/share/info/
+      done
     '';
   });
   hledger-ui = overrideCabal super.hledger-ui (drv: {
     postInstall = ''
       for i in $(seq 1 9); do
-        for j in $data/share/${self.ghc.name}/${pkgs.stdenv.system}-${self.ghc.name}/*/*.$i $data/share/${self.ghc.name}/${pkgs.stdenv.system}-${self.ghc.name}/*/.otherdocs/*.$i; do
+        for j in $data/share/${self.ghc.name}/${ghc_system}-${self.ghc.name}/*/*.$i $data/share/${self.ghc.name}/${ghc_system}-${self.ghc.name}/*/.otherdocs/*.$i; do
           mkdir -p $out/share/man/man$i
           cp $j $out/share/man/man$i/
         done
       done
-      mkdir $out/share/info
-      cp $data/share/${self.ghc.name}/${pkgs.stdenv.system}-${self.ghc.name}/*/*.info $out/share/info/
+      for i in $data/share/${self.ghc.name}/${ghc_system}-${self.ghc.name}/*/*.info; do
+        mkdir -p $out/share/info
```

Comment:
Why didn't you just replace the old `mkdir $out/share/info` with `mkdir -p $out/share/info`?
The behaviour should be the same as this loop.